pages/auth/index.js

```diff
@@ -1,26 +1,299 @@
 import { useState, useEffect } from 'react'
-import { Button } from '@material-ui/core'
-import { signIn, signOut, getSession } from 'next-auth/client'
+import { getProviders, signIn, signOut, getSession } from 'next-auth/client'
+import Image from 'next/image'
+import axios from 'axios'
+import Button from '@material-ui/core/Button'
+import TextField from '@material-ui/core/TextField'
+import Dialog from '@material-ui/core/Dialog'
+import {
+  Box,
+  Divider,
+  Grid,
+  Hidden,
+  Typography,
+  IconButton,
+} from '@material-ui/core'
 
-function Auth() {
+import styles from '../../styles/Auth/Auth.module.css'
+
+const Auth = ({ providers }) => {
   const [loading, setLoading] = useState(true)
   const [session, setSession] = useState()
+  const [modalOpen, setModalOpen] = useState(false)
+  const [isSignUp, setIsSignUp] = useState(false)
+  const [firstName, setFirstName] = useState('')
+  const [lastName, setLastName] = useState('')
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [validFirstName, setValidFirstName] = useState(false)
+  const [validLastName, setValidLastName] = useState(false)
+  const [validEmail, setValidEmail] = useState(false)
+  const [validPassword, setValidPassword] = useState(false)
+  const [submitted, setSubmitted] = useState(false)
+
+  const AuthButton = () => {
+    if (session) {
+      return (
+        <Button variant="outlined" color="primary" onClick={() => signOut()}>
+          Signout
+        </Button>
+      )
+    }
+    return (
+      <Button
+        variant="outlined"
+        color="primary"
+        onClick={() => setModalOpen(true)}
+      >
+        {isSignUp ? 'Login' : 'Signup'}
+      </Button>
+    )
+  }
+
+  const validSignUp =
+    validFirstName && validLastName && validEmail && validPassword
+  const validLogIn = validEmail && validPassword
+
+  const validateFirstName = (value) => {
+    setValidFirstName(/^[a-z ,.'-]+$/i.test(value))
+  }
+
+  const validateLastName = (value) => {
+    setValidLastName(/^[a-z ,.'-]+$/i.test(value))
+  }
+
+  const validateEmail = (value) => {
+    setValidEmail(
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(
+        value
+      )
+    )
+  }
+
+  const validatePassword = (value) => {
+    setValidPassword(
+      !isSignUp ||
+        /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/.test(
```

Comment:
what was ur criteria for validation? Are characters like `,.'- ` necessary for names?

Reply:
Hello Jimmy D'Housse-Austin :D